src/Decompiler/Evaluation/IdConstant.cs

```diff
@@ -71,10 +71,11 @@ public Expression Transform()
             if (this.pt != null)
             {
                 ctx.RemoveIdentifierUse(idDst!);
+                var pt = idDst!.DataType.ResolveAs<PrimitiveType>();
                 var cNew = src!.CloneExpression();
                 if (src.DataType.IsWord &&
-                    src is Constant cSrc && 
-                    idDst!.DataType is PrimitiveType pt &&
+                    src is Constant cSrc &&
+                    pt != null &&
```

Comment:
This looks good. Just a stylistic note: let's start using `is null` and `is not null` instead of `== null` and `!= null`. The `is (not) null` operator will never call `operator ==` if a class has it defined. Not necessary for this PR, just keep it in mind.